src/components/NcButton/NcButton.vue

```diff
@@ -415,308 +430,314 @@ td.row-size {
+	emit('click', event)
+	routerLink.value?.navigate(event)
+}
+</script>
+
+<template>
+	<component :is="isLink ? 'a' : 'button'"
+		class="button-vue"
+		:class="[
+			`button-vue--size-${size}`,
 			{
-				class: [
-					'button-vue',
-					`button-vue--size-${this.size}`,
-					{
-						'button-vue--icon-only': hasIcon && !hasText,
-						'button-vue--text-only': hasText && !hasIcon,
-						'button-vue--icon-and-text': hasIcon && hasText,
-						[`button-vue--vue-${this.realType}`]: this.realType,
-						'button-vue--wide': this.wide,
-						[`button-vue--${this.flexAlignment}`]: this.flexAlignment !== 'center',
-						'button-vue--reverse': this.isReverseAligned,
-						active: isActive,
-					},
-				],
-				'aria-label': this.ariaLabel,
-				'aria-pressed': hasPressed ? this.pressed.toString() : undefined,
-				disabled: this.disabled,
-				type: isLink ? null : this.nativeType,
-				role: isLink ? 'button' : null,
-				href: this.to ? href : (this.href || null),
-				target: isLink ? this.target : null,
-				rel: isLink ? 'nofollow noreferrer noopener' : null,
-				download: (!this.to && this.href && this.download) ? this.download : null,
-				// If this button is used as a popover trigger, we need to apply trigger attrs, e.g. aria attributes
-				...this.ncPopoverTriggerAttrs,
-				onClick: ($event) => {
-					// Update pressed prop on click if it is set
-					if (hasPressed) {
-						/**
-						 * Update the current pressed state of the button (if the `pressed` property was configured)
-						 *
-						 * @property {boolean} newValue The new `pressed`-state
-						 */
-						this.$emit('update:pressed', !this.pressed)
-					}
-					// We have to both navigate and emit the click event
-					this.$emit('click', $event)
-					navigate?.($event)
-				},
+				[`button-vue--${realVariant}`]: realVariant,
+				'button-vue--wide': wide,
+				[`button-vue--${flexAlignment}`]: flexAlignment !== 'center',
+				'button-vue--reverse': isReverseAligned,
+				active: routerLink?.isActive,
 			},
-			[
-				h('span', { class: 'button-vue__wrapper' }, [
-					hasIcon
-						? h('span', {
-							class: 'button-vue__icon',
-							'aria-hidden': 'true',
-						},
-						[this.$slots.icon?.()],
-						)
-						: null,
-					hasText ? h('span', { class: 'button-vue__text' }, [this.$slots.default?.()]) : null,
-				]),
-			],
-		)
-
-		// If we have a router-link, we wrap the button in it
-		if (this.to) {
-			return h(resolveComponent('router-link'), {
-				custom: true,
-				to: this.to,
-			}, {
-				default: renderButton,
-			})
-		}
-		// Otherwise we simply return the button
-		return renderButton()
-	},
-})
-</script>
+		]"
+		:disabled
+		:aria-label
+		v-bind="{
+			...ncPopoverTriggerAttrs,
+			...(isLink ? linkAttrs : buttonAttrs),
+		}"
+		@click="onClick">
+		<span class="button-vue__wrapper">
+			<span v-if="hasIcon()" class="button-vue__icon">
+				<slot name="icon" />
+			</span>
+			<span v-if="hasText()" class="button-vue__text">
```

Comment:
I'm still not fun of x2 rendering 👀

Reply:
And it is not really compatible with vapor mode

Reply:
see fixup commit to make it css only